src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/set_interior_vehicle_data_request.cc

```diff
@@ -221,9 +359,27 @@ void SetInteriorVehicleDataRequest::Execute() {
     (*message_)[app_mngr::strings::msg_params][app_mngr::strings::app_id] =
         app->app_id();
 
+    const bool app_wants_to_set_audio_src =
+        module_data.keyExists(message_params::kAudioControlData) &&
+        module_data[message_params::kAudioControlData].keyExists(
+            message_params::kSource);
+
+    if (app_wants_to_set_audio_src && !app->IsAllowedToChangeAudioSource()) {
+      LOG4CXX_WARN(logger_, "App is not allowed to change audio source");
+      SetResourceState(ModuleType(), ResourceState::FREE);
```

Comment:
why need SetResourceState(ModuleType(), ResourceState::FREE);?

Reply:
@yang1070 because we execute the command only if the resource successfully acquired https://github.com/smartdevicelink/sdl_core/blob/feature/rc_new_modules/src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/rc_command_request.cc#L144

and release it when we received response from HMI 
https://github.com/smartdevicelink/sdl_core/blob/feature/rc_new_modules/src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/rc_command_request.cc#L192

so in case when smthg goes wrong (when we execute the command) and we did not send a request to HMI, we need to release the resource before sending a response to mobile.

Reply:
Let consider the scenario
app1 set the volume, it succeeds, AUDIO module is assigned to app1
app1 set some value of equalizer,  it succeeds, AUDIO module is still assigned to app1
app1 tried to set the audio source, and does not pass the check of line 374, AUDIO shall still be allocated to app1, AUDIO is not free.

Reply:
@yang1070 AUDIO module is still assigned to app1, than app2 will received response "IN_USE" from SDL, in
https://github.com/smartdevicelink/sdl_core/blob/feature/rc_new_modules/src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/rc_command_request.cc#L170

and SDL will not execute command for app2 because of this check 
https://github.com/smartdevicelink/sdl_core/blob/feature/rc_new_modules/src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/rc_command_request.cc#L144

Reply:
@yang1070
I tested this case, so in current implementation:
1. app1 set the volume, it succeeds, AUDIO module is assigned to app1 (SDL don't receive response from HMI) 
2. app1 set some value of equalizer and receive from SDL response "IN_USE" 

then when we receive response from HMI for step 1 -> AUDIO module is released 

3. app1 tried to set the audio source (and acquire AUDIO module), and does not pass the check of line 374 -> AUDIO module becomes free.

maybe we should change this logic, but actually this is an old RC functionality, here is a link for our requirements for resource allocation
https://github.com/smartdevicelink/sdl_requirements/blob/master/detailed_docs/RC/resource_allocation.md